crates/aptos/src/move_tool/show.rs

```diff
@@ -0,0 +1,109 @@
+        }
+    }
+}
+
+/// Compile the package and show information about the ABIs of the compiled modules.
+///
+/// For example, this would show the function `transfer` in the module `coin`:
+///
+/// aptos move show abi --modules coin --names transfer
+///
+#[derive(Parser)]
+pub struct ShowAbi {
+    /// If provided, only show items from the given Move modules. These should be module
+    /// names, not file paths. For example, `coin`.
+    #[clap(long, multiple_values = true)]
+    modules: Vec<String>,
+
+    /// If provided, only show items with the given names. For example, `transfer`.
+    #[clap(long, multiple_values = true)]
+    names: Vec<String>,
+
+    #[clap(flatten)]
+    included_artifacts_args: IncludedArtifactsArgs,
+
+    #[clap(flatten)]
+    move_options: MovePackageDir,
+}
+
+#[async_trait]
+impl CliCommand<Vec<EntryABI>> for ShowAbi {
+    fn command_name(&self) -> &'static str {
+        "ShowAbi"
+    }
+
+    async fn execute(self) -> CliTypedResult<Vec<EntryABI>> {
+        let build_options = BuildOptions {
+            install_dir: self.move_options.output_dir.clone(),
+            with_abis: true,
+            ..self
+                .included_artifacts_args
+                .included_artifacts
+                .build_options(
+                    self.move_options.skip_fetch_latest_git_deps,
+                    self.move_options.named_addresses(),
+                    self.move_options.bytecode_version,
+                )
+        };
+
+        // Build the package.
+        let package = BuiltPackage::build(self.move_options.get_package_path()?, build_options)
+            .map_err(|e| CliError::MoveCompilationError(format!("{:#}", e)))?;
+
+        // Get ABIs from the package.
+        let abis = package
+            .extract_abis()
+            .context("No ABIs found after compilation")?;
+
+        // Filter the ABIs based on the filters passed in.
+        let abis = abis
+            .into_iter()
+            .filter(|abi| {
+                let name = abi.name().to_string();
+                if !self.names.is_empty() && !self.names.contains(&name) {
+                    return false;
+                }
+                match &abi {
+                    EntryABI::EntryFunction(func) => {
+                        if !self.modules.is_empty()
+                            && !self
+                                .modules
+                                .contains(&func.module_name().name().to_string())
+                        {
+                            return false;
+                        }
+                    },
+                    EntryABI::TransactionScript(_) => {
+                        // If there were any modules specified we ignore scripts.
+                        if !self.modules.is_empty() {
+                            return false;
+                        }
+                    },
+                }
+                true
```

Comment:
```suggestion
                match &abi {
                    EntryABI::EntryFunction(func) => {
                        !(self.modules.is_empty()
                            || self
                                .modules
                                .contains(&func.module_name().name().to_string()))
                    },
                    EntryABI::TransactionScript(_) => {
                        // If there were any modules specified we ignore scripts.
                        !self.modules.is_empty()
                    },
                    _ => true
                }
```

the code can be slimmed down a bit

Reply:
Doesn't vibe with my brain as well, I like the series of `check -> return false` parts followed by a final `true`. But happy to make the change.